src/command-helpers/spinner.js

```diff
@@ -14,19 +14,27 @@ const step = (spinner, subject, text) => {
   return spinner
 }
 
-// A wrapper for functions (f) that provides a progress spinner and formatted alert messages.
-// For warning message support the f must return an immutable Map with result and warning keys.
+// Executes the function `f` in a command-line spinner, using the
+// provided captions for in-progress, error and failed messages.
+//
+// If `f` throws an error, the spinner stops with the failure message
+//   and rethrows the error.
+// If `f` returns an object with a `warning` and a `result` key, the
+//   spinner stops with the warning message and returns the `result` value.
+// Otherwise the spinner prints the in-progress message with a check mark
+//   and simply returns the value returned by `f`.
 const withSpinner = async (text, errorText, warningText, f) => {
   let spinner = toolbox.print.spin(text)
   try {
     let result = await f(spinner)
-    if(warningText && immutable.Map.isMap(result)) {
-      if (result.get('warning', false)) {
-        spinner.stopAndPersist({symbol: '⚠️', text: `${warningText}: ${result.get('warning')}`})
-        return result.get('result')
+    if (typeof result === 'object') {
+      if (result.warning && Object.keys(result).indexOf('result') >= 0) {
+        spinner.warn(`${warningText}: ${result.warning}`)
+        spinner.succeed(text)
```

Comment:
Is it necessary to have both `spinner.warn` and `spinner.succeed` when there is a warning?

Reply:
Yep, because we want to see `[x] Load subgraph` even if there were warnings.

Reply:
My understanding is that the loading text (in this case "Load subgraph from...") is set by the first line, `let spinner = toolbox.print.spin(text)`.   I don't see any difference in the output with the line in question removed.

Reply:
First with the extra `succeed`, then without it:
![image](https://user-images.githubusercontent.com/19324/54529089-2f9cf300-497f-11e9-86ef-9f3ed899edde.png)

Only in the first example is `✔ Load subgraph from subgraph.yaml` logged after the warnings. And that's what we want.